encoder/vaapiencoder_h264.cpp

```diff
@@ -1516,21 +1627,60 @@ bool VaapiEncoderH264::fill(VAEncPictureParameterBufferH264* picParam, const Pic
     picParam->pic_fields.bits.transform_8x8_mode_flag = m_videoParamAVC.enableDct8x8;
     picParam->pic_fields.bits.deblocking_filter_control_present_flag = true;
 
+    picParam->pic_fields.bits.pic_scaling_matrix_present_flag =
+        m_videoParamAVC.qMatrix.picScalingMatrixPresentFlag;
+
     return TRUE;
 }
 
+bool VaapiEncoderH264::fill(VAIQMatrixBufferH264* buffer) const
+{
+    uint8_t *dest = (uint8_t *)buffer;
+    const VideoParamsQM *qMatrix = &m_videoParamAVC.qMatrix;
+    const uint8_t *list_present_flag;
+    int i;
+
+    if (qMatrix->picScalingMatrixPresentFlag) {
+        list_present_flag = qMatrix->picScalingListPresentFlag;
+    } else if (qMatrix->seqScalingMatrixPresentFlag) {
+        list_present_flag = qMatrix->seqScalingListPresentFlag;
+    } else {
+        return true;
```

Comment:
This will send random value to driver since you did not fill the buffer.right?So this should not happen. Since you have checked value in 1251, you have two option
1. add a list_present_flag param to VaapiEncoderH264::fill(VAIQMatrixBufferH264* buffer), I have kind of prefer this.
2.  assert at 1754.  it will make code more clearer.